llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -1096,6 +1096,50 @@ Value *InstCombinerImpl::foldUsingDistributiveLaws(BinaryOperator &I) {
   return SimplifySelectsFeedingBinaryOp(I, LHS, RHS);
 }
 
+bool InstCombinerImpl::matchSymmetricPhiNodesPair(PHINode *LHS, PHINode *RHS) {
+
+  if (LHS->getNumIncomingValues() != 2 || RHS->getNumIncomingValues() != 2)
+    return false;
+
+  BasicBlock *B0 = LHS->getIncomingBlock(0);
+  BasicBlock *B1 = LHS->getIncomingBlock(1);
+
+  bool RHSContainB0 = RHS->getBasicBlockIndex(B0) != -1;
+  bool RHSContainB1 = RHS->getBasicBlockIndex(B1) != -1;
+
+  if (!RHSContainB0 || !RHSContainB1)
```

Comment:
Please check that the phi nodes are in the same basic block instead.

Reply:
Does this mean we need to add a check to see if the phi nodes are in the same basic block?

Reply:
Yes, check that `LHS->getParent() == RHS->getParent()`.